### DISCUSSION TITLE
Implement correct display of vector-like structs

### DISCUSSION THREAD

**Discussion by haakon-e** [fonsp/Pluto.jl#2936] [Questions about using Pluto]
Hi there,

I'm wondering if there's a "template" or suggested approach for how to make vector-like (or row-like) structs display "nicely" in Pluto.

This question was originally asked in https://github.com/rafaqz/DimensionalData.jl/issues/725

Some key information from that post:
------
Printing `DimArray`s works nicely in (VSCode) terminal window:
![image](https://github.com/rafaqz/DimensionalData.jl/assets/45243236/1f8095dd-beef-4430-a943-0f2a9a4e0aff)

but in Pluto, it only works for `DimArray`s with multiple dimensions:
<img width="730" alt="image" src="https://github.com/rafaqz/DimensionalData.jl/assets/45243236/04db464d-826e-4075-819f-efbddad97793">

**Comment by greimel** [comment]
The example above doesn't make clear why better printing is useful.

The package allows to construct arrays that with named dimensions. E.g.

```julia-repl
julia> temperature = DimArray(round.(rand(5) .* 30, digits=1), (city = ["Vienna", "Berlin", "Prague", "Bratislava", "Budapest"],))
╭───────────────────────────────╮
│ 5-element DimArray{Float64,1} │
├───────────────────────────────┴──────────────────────────────────────── dims ┐
  ↓ city Categorical{String} ["Vienna", "Berlin", "Prague", "Bratislava", "Budapest"] Unordered
└──────────────────────────────────────────────────────────────────────────────┘
 "Vienna"      24.4
 "Berlin"       1.5
 "Prague"       3.0
 "Bratislava"  18.1
 "Budapest"    28.9

julia> temperature[city=At("Vienna")]
24.4
```

Would be nice if Pluto could display the correct labels, at least when the vector is unfolded

![image](https://github.com/user-attachments/assets/dda7e96c-d446-4e43-82b7-c6e5c72d1152)

**Comment by fonsp** [comment]
Thanks for the feedback!

Perhaps the Pluto tree viewer should be disabled for all `DimArray`s? @greimel What do you think?

If so, we can add an exception similar to Symbolics.jl here https://github.com/fonsp/Pluto.jl/blob/aaa835ba8ba3318f46231af354fb58dd29cc91c7/src/runner/PlutoRunner/src/integrations.jl#L43-L48 (feel free to make a PR!)

**Reply by fonsp** [reply]
@haakon-e Could you make a PR?
